### DISCUSSION TITLE
Great Work

### DISCUSSION THREAD

**Discussion by lohtse** [egzumer/uv-k5-firmware-custom#156] [General]
Now as most  know already.. the two of us do NOT get along at all...   But I do believe that even when that is the case that when praise is due you should still pass it on... 

Up until yesterday(before [59b3251) I would often have too make compromises when building a rom from you(and others).. and then you added more features which would result in even bigger FW and more has to be disabled. BUT as of release/Commit 59b3251 for the first time ever I could enable everything I always wanted  and more!!   That is outstanding and shows how much you have refactored, Rewrote and optimised the code  since all previous builds... I was actually astonished... so much so I deleted the folder  redownloaded and rebuilt to make sure...   And the same results...  

You have done amazing work on the FW and for that I am grateful and appreciate the work you have put in...

Thank you 

regards
Lohtse

**Comment by egzumer** [comment]
Don't get too excited, you probably enabled NOAA, which in turn disables a call to spectrum, which causes it to be effectively not used at all and optimised out of the binary 😀
There are no free lunches.

**Reply by lohtse** [reply]
Errrrm Nope NOAA is not enabled, I never enable it and always check what I have enabled and disabled before going forward.. And spectrum is work fine too.

**Reply by egzumer** [reply]
Well then, you welcome. There were some optimisations, but in general they were consumed by other modification, but if it works for you then greate.

**Reply by lohtse** [reply]
Just checked

Checked all functions that I enabled which I normally cannot and they all work...   


What ever you did really did a great job on allowing more.... 
![image](https://github.com/egzumer/uv-k5-firmware-custom/assets/48331155/2ee7ebf1-fc54-4193-b2e4-d3dc8cbe5085)
thats what I have enabled,,

If I disable stuff the FW size will reduce as it should and same when adding..

So Agian

THANK YOU for hardwork

**Reply by egzumer** [reply]
Ahh yes, you disabled DTMF calling, that is an option that I added yesturday. Not a result of optimisation really, but an extra configuration option 👍
